base/libgit2/oid.jl

```diff
@@ -71,12 +71,12 @@ Base.string(id::Oid) = hex(id)
 
 Base.show(io::IO, id::Oid) = print(io, "Oid($(string(id)))")
 
-Base.hash(id::Oid) = hash(id.val)
+Base.hash(id::Oid, h::UInt) = hash(id.val, h)
 
-cmp(id1::Oid, id2::Oid) = int(ccall((:git_oid_cmp, :libgit2), Cint,
-                                         (Ptr{Oid}, Ptr{Oid}), Ref(id1), Ref(id2)))
```

Comment:
We currently use the bytestring SHA's everywhere, but that should be transitioned over to using Oid's for some type safety.